subliminal/providers/legendastv.py

```diff
@@ -0,0 +1,466 @@
+            r = self.session.get('%s/users/logout' % self.server_url, timeout=TIMEOUT)
+            r.raise_for_status()
+            logger.debug('Logged out')
+            self.logged_in = False
+
+        self.session.close()
+
+    def matches(self, expected, actual, ignore_episode=False):
+        """
+        Matches two dictionaries (expected and actual). The dictionary keys follow the guessit properties names.
+        If the expected dictionary represents a movie:
+          - ``type`` should match
+          - ``title`` should match
+          - ``year`` should match, unless they're not defined and expected and actual ``title``s are the same
+        If the expected dictionary represents an episode:
+          - ``type`` should match
+          - ``series`` should match
+          - ``eason`` should match
+          - ``episodeNumber`` should match, unless ``ignore_episode`` is True
+
+        :param expected: dictionary that contains the expected values
+        :param actual: dictionary that contains the actual values
+        :param ignore_episode: True if should ignore episodeNumber matching. Default: False
+        :return: True if actual matches expected
+        :rtype: bool
+        """
+        if expected.get('type') != actual.get('type'):
+            return False
+
+        if expected.get('type') == 'movie':
+            if not self.name_matches(expected.get('title'), actual.get('title')):
+                return False
+            if expected.get('year') != actual.get('year'):
+                if expected.get('year') and actual.get('year'):
+                    return False
+                if expected.get('title', '').lower() != actual.get('title', '').lower():
+                    return False
+
+        elif expected.get('type') == 'episode':
+            if not self.name_matches(expected.get('series'), actual.get('series')):
+                return False
+            if expected.get('season') != actual.get('season'):
+                return False
+            if not ignore_episode and expected.get('episodeNumber') != actual.get('episodeNumber'):
+                return False
+
+        return True
+
+    def name_matches(self, expected_name, actual_name):
+        """
+        Returns True if expected name and actual name matches. To be considered a match, actual_name should contain
+        all words of expected_name in the order of appearance.
+
+        :param expected_name:
+        :param actual_name:
+        :return: True if actual_name matches expected_name
+        :rtype : bool
+        """
+        if not expected_name or not actual_name:
+            return expected_name == actual_name
+
+        words = self.word_split_re.findall(expected_name)
+        name_regex_re = re.compile('(.*' + '\W+'.join(words) + '.*)', re.IGNORECASE)
+
+        return name_regex_re.match(actual_name)
+
+    @region.cache_on_arguments(expiration_time=SHOW_EXPIRATION_TIME)
+    def search_titles(self, params):
```

Comment:
I'd prefer explicit args rather than implicit ones. Also I think you can decouple from the `params` logic for this function. Functions in general should be simple to understand, simple args, simple return value.